src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapper.java

```diff
@@ -1,18 +1,134 @@
+        // Observation.status = SG18.STS.C555.9011
+        labResult.getTestStatus()
+            .map(TestStatus::getTestStatusCode)
+            .map(TestStatusCode::getDescription)
+            .map(String::toLowerCase)
+            .map(ObservationStatus::fromCode)
+            .ifPresent(observation::setStatus);
+    }
+
+    private void mapValueQuantity(final LabResult labResult, final Observation observation) {
+        // Observation.value.valueQuantity.*
+        labResult.getInvestigationResult().ifPresent(investigationResult -> {
+            final var quantity = new Quantity();
+
+            // Observation.value.valueQuantity.value = SG18.RSL.C830(1).6314
+            quantity.setValue(investigationResult.getMeasurementValue());
+
+            // Observation.value.valueQuantity.unit = SG18.RSL.C848.6410
+            quantity.setUnit(investigationResult.getMeasurementUnit());
+
+            // Observation.value.valueQuantity.comparator = SG18.RSL.C830(1).6321
+            investigationResult.getMeasurementValueComparator()
+                .map(MeasurementValueComparator::getDescription)
+                .map(QuantityComparator::fromCode)
+                .ifPresent(quantity::setComparator);
+
+            observation.setValue(quantity);
+        });
+    }
+
+    private void mapCode(final LabResult labResult, final Observation observation) {
+        // Observation.code = SG18.INV.C847.9930 and SG18.INV.C847.9931
+        observation.getCode().setText(labResult.getInvestigation().getInvestigationDescription());
+        observation.getCode().addCoding().setCode(labResult.getInvestigation().getInvestigationCode());
+    }
+
+    private void mapReferenceRange(final LabResult labResult, final Observation observation) {
+        // Observation.referenceRange.*
+        labResult.getResultReferenceRanges().stream()
+            .map(ResultReferenceRange::getDetails)
+            .map(rangeDetail -> {
+                final var range = new ObservationReferenceRangeComponent();
+
+                // Observation.referenceRange.low = SG20.RND.6162
+                rangeDetail.getLowerLimit()
+                    .map(this::toSimpleQuantity)
+                    .ifPresent(range::setLow);
+
+                // Observation.referenceRange.high = SG20.RND.6152
+                rangeDetail.getUpperLimit()
+                    .map(this::toSimpleQuantity)
+                    .ifPresent(range::setHigh);
+
+                return range;
+            })
+            .forEach(observation::addReferenceRange);
+    }
+
+    private void mapComment(final LabResult labResult, final Observation observation) {
+        // Observation.comment = SG18.FTX.C108.4440(1-5)
+        Optional.of(labResult.getFreeTexts().stream()
+            .map(FreeTextSegment::getTexts)
+            .map(texts -> String.join(" ", texts))
+            .collect(Collectors.joining("\n")))
+            .filter(StringUtils::isNotBlank)
```

Comment:
Is this needed? The `FreeTextSegment` parser ensures there are no blanks

Reply:
According to the word doc, we could have a FTX segment with no data at all, indicating a blank line to the consuming service. So, I'll need to make this preserve that somehow

Reply:
Right, so this filter is actually just making sure that after the full FHIR comment has been compiled, it actually has some content. It stops us setting a content of only whitespace.

I've updated a test to include a blank line.